Include/internal/pycore_object.h

```diff
@@ -32,6 +32,16 @@ PyAPI_FUNC(void) _Py_NO_RETURN _Py_FatalRefcountErrorFunc(
 #define _Py_FatalRefcountError(message) \
     _Py_FatalRefcountErrorFunc(__func__, (message))
 
+// Increment reference counter by n
+static inline void _Py_RefcntAdd(PyObject* op, Py_ssize_t n)
+{
+#ifdef Py_REF_DEBUG
+    _Py_RefTotal += n;
+#endif
+    op->ob_refcnt += n;
```

Comment:
I would prefer to use `Py_SET_REFCNT(op, Py_REFCNT(op) + n)` in case something like https://github.com/python/cpython/pull/19474 is ever applied.

Reply:
On second thought, this requires care either way, so it should be fine to leave it as `op->ob_refcnt`.